aqt/archives.py

```diff
@@ -36,18 +38,40 @@ def __init__(self, version, target, arch, os_name):
         self.arch = arch
         self.os_name = os_name
 
+    def __str__(self):
+        print(f'TargetConfig(version={self.version}, target={self.target}, '
+              f'arch={self.arch}, os_name={self.os_name}')
+
+    def __repr__(self):
+        print(f'({self.version}, {self.target}, {self.arch}, {self.os_name})')
+
 
 class QtPackage:
     """
     Hold package information.
     """
 
-    def __init__(self, name, archive_url, archive, package_desc, hashurl):
+    def __init__(self, name: str , archive_url: str, archive: str,
+                 package_desc: str, hashurl: str,
+                 version: Optional[Version] = None):
         self.name = name
         self.url = archive_url
         self.archive = archive
         self.desc = package_desc
         self.hashurl = hashurl
+        self.version = version
```

Comment:
I store a version on the package. 
In the case of tools, there might be several matches. We want to filter on the max version to return a single